testing/integration/ess/repackage.go

```diff
@@ -0,0 +1,297 @@
+	// starting from SNAPSHOT packages, so we have to work around the fact that we cannot simply re-generate the packages
+	// by defining versions in 2 separate ways for repackage hack and for fixtures
+	buildMetadataForAgentFixture := newVersionBuildMetadata
+	if currentVersion.IsSnapshot() {
+		buildMetadataForAgentFixture += "-SNAPSHOT"
+	}
+	versionForFixture := version.NewParsedSemVer(currentVersion.Major(), currentVersion.Minor(), currentVersion.Patch(), "", buildMetadataForAgentFixture)
+
+	// calculate the new package name
+	newPackageFileName := strings.Replace(originalPackageFileName, currentVersion.String(), versionForFixture.String(), 1)
+	t.Logf("originalPackageName: %q newPackageFileName: %q", originalPackageFileName, newPackageFileName)
+
+	newPackageAbsPath := filepath.Join(newPackageContainingDir, newPackageFileName)
+
+	// hack the package based on type
+	ext := filepath.Ext(originalPackageFileName)
+	if ext == ".gz" {
+		// fetch the next extension
+		ext = filepath.Ext(strings.TrimRight(originalPackageFileName, ext)) + ext
+	}
+	switch ext {
+	case ".zip":
+		t.Logf("file %q is a .zip package", originalPackageFileName)
+		repackageZipArchive(t, srcPackage, newPackageAbsPath, parsedNewVersion)
+	case ".tar.gz":
+		t.Logf("file %q is a .tar.gz package", originalPackageFileName)
+		repackageTarArchive(t, srcPackage, newPackageAbsPath, parsedNewVersion)
+	default:
+		t.Logf("unknown extension %q for package file %q ", ext, originalPackageFileName)
+		t.FailNow()
+	}
+
+	// Create hash file for the new package
+	err = mage.CreateSHA512File(newPackageAbsPath)
+	require.NoErrorf(t, err, "error creating .sha512 for file %q", newPackageAbsPath)
+	return versionForFixture, err
+}
+
+func repackageTarArchive(t *testing.T, srcPackagePath string, newPackagePath string, newVersion *version.ParsedSemVer) {
+	oldTopDirectoryName := strings.TrimRight(filepath.Base(srcPackagePath), ".tar.gz")
+	newTopDirectoryName := strings.TrimRight(filepath.Base(newPackagePath), ".tar.gz")
+
+	// Open the source package and create readers
+	srcPackageFile, err := os.Open(srcPackagePath)
+	require.NoErrorf(t, err, "error opening source file %q", srcPackagePath)
+	defer func(srcPackageFile *os.File) {
+		err := srcPackageFile.Close()
+		if err != nil {
+			assert.Failf(t, "error closing source file %q: %v", srcPackagePath, err)
+		}
+	}(srcPackageFile)
+
+	gzReader, err := gzip.NewReader(srcPackageFile)
+	require.NoErrorf(t, err, "error creating gzip reader for file %q", srcPackagePath)
+	defer func(gzReader *gzip.Reader) {
+		err := gzReader.Close()
+		if err != nil {
+			assert.Failf(t, "error closing gzip reader for source file %q: %v", srcPackagePath, err)
+		}
+	}(gzReader)
+
+	tarReader := tar.NewReader(gzReader)
+
+	// Create the output file and its writers
+	newPackageFile, err := os.OpenFile(newPackagePath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0o750)
+	require.NoErrorf(t, err, "error opening output file %q", newPackageFile)
```

Comment:
%q with a *File, won't look that well right? 🙂

Reply:
Fixed in 98d4f2b159dbf99dc1a9cfc6a291d7cf6ca50693